Info-Orbs/src/widgets/webDataWidget.cpp

```diff
@@ -0,0 +1,158 @@
+
+#include "widgets/webDataWidget.h"
+
+#include <config.h>
+#include <globalTime.h>
+
+WebDataWidget::WebDataWidget(ScreenManager &manager) : Widget(manager) {
+    m_lastUpdate = 0;
+}
+
+WebDataWidget::~WebDataWidget() {
+}
+
+void WebDataWidget::setup() {
+}
+
+void WebDataWidget::draw() {
+    draw(false);
+}
+
+void WebDataWidget::draw(bool force) {
+    uint32_t background = TFT_BLACK;
+    uint32_t color = TFT_CYAN;
+
+    // Weather, displays a clock, city & text weather discription, weather icon, temp, 3 day forecast
+    for (int i=0 ; i<5; i++) {
+        String fingerprint = getFingerprint(i);
+        if (fingerprint != m_fingerprint[i]) {
+            m_manager.selectScreen(i);
+
+            TFT_eSPI &display = m_manager.getDisplay();
+            display.fillScreen(m_background[i]);
+            display.setTextColor(m_labelColor[i]);
+            display.setTextSize(2);
+            display.setTextDatum(MC_DATUM);
+            display.drawString(m_label[i], 120, 75, 2);
+            display.setTextColor(m_color[i]);
+            display.setTextDatum(MC_DATUM);
+            if (m_data[i].length()>10) {
```

Comment:
we should also check the string for newlines and break them up. But with variable font sizes x,y positioning becomes tricky